src/haddock/modules/analysis/ilrmsdmatrix/__init__.py

```diff
@@ -212,6 +212,13 @@ def _run(self) -> None:
             path=contact_obj.path,
             ncores=ncores
             )
+        # if the chains in res_resdic are empty, then there are no contacts and
+        # the ilrmsd matrix cannot be calculated
+        for ch in res_resdic:
+            if res_resdic[ch].size == 0:
```

Comment:
Are you sure of this check ?
Maybe one can define `ligand_chains = ['B', 'C']` and chain 'C' do not contain contacting residues but chain 'B' yes ?